feat(curriculum): forum leaderboard project (#50241)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-fetch-promises-and-async-await-by-building-an-fcc-forum-leaderboard/644fa585a5f9970173ca3cff.md

---
id: 644fa585a5f9970173ca3cff
title: Step 77
challengeType: 0
dashedName: step-77
---

# --description--

> Create a constant named `linkClass`, and assign 2 things in a template literal, first the word `category` and then the property `selectedCategory.className`. Both of them are class names that are used to apply styles to the anchor tag.

Reviewer: ```suggestion
Create a constant named `linkClass`, and assign 2 things in a template literal, first the word `category`, followed by a space, and then the property `selectedCategory.className`. Both of them are class names that are used to apply styles to the anchor tag.
```

---
id: 644fa585a5f9970173ca3cff
title: Step 57
challengeType: 0
dashedName: step-57
---

# --description--

> Create a constant named `linkClass`, and assign 2 things in a template literal, first the word `category`, followed by a space, and then the property `selectedCategory.className`. Both of them are class names that are used to apply styles to the anchor tag.

Reviewer 2: ```suggestion
Create a constant named `linkClass` and assign it a template literal. Inside that template literal, add the word `category` followed by a space, and then an embedded expression for `selectedCategory.className`. 

These class names will be used to apply styles for the anchor element.
```

# --hints--

You should create a constant named `linkClass` and assign a template literal to it.

```js
assert(code.match(/const\s+linkClass\s*=\s*`/g));
```

> You should assign the word `category` to the template literal.

Reviewer 2: ```suggestion
You should have the word `category` in the template literal.
```

```js
assert(code.match(/const\s+linkClass\s*=\s*`\s*category.*`/g));
```

> You should assign property `selectedCategory.className` after the word `category` inside the template literal using a dollar sign and curly braces.

Reviewer 2: ```suggestion
You should have a space after the word `category` and then have an embedded expression for `selectedCategory.className`. 
```

  const linkText = selectedCategory.category;
--fcc-editable-region--

--fcc-editable-region--
};

const timeAgo = (time) => {
  const currentTime = new Date();
  const lastPost = new Date(time);

  const timeDifference = currentTime - lastPost;
  const MS_PER_MINUTE = 1000 * 60;

  const minutesAgo = Math.floor(timeDifference / MS_PER_MINUTE);
  const hoursAgo = Math.floor(minutesAgo / 60);
  const daysAgo = Math.floor(hoursAgo / 24);

  if (minutesAgo < 60) {
    return `${minutesAgo}m ago`;
  }

  if (hoursAgo < 24) {
    return `${hoursAgo}h ago`;
  }

  return `${daysAgo}d ago`;
};

const viewCount = (views) => {
  const thousands = Math.floor(views / 1000);

  if (views >= 1000) {
    return `${thousands}k`;
  }

  return views;
};

const fetchData = async () => {
  try {
    const res = await fetch(FORUM_LATEST);
    const data = await res.json();
    console.log(data);
    showLatestPosts(data);
  } catch (err) {
    console.log(err);
  }
};

fetchData();

const showLatestPosts = (data) => {
  const { topic_list, users } = data;
  const { topics } = topic_list;

  postsContainer.innerHTML = topics.map((item) => {

>     const {
>       id,
>       title,
>       views,
>       posts_count,
>       slug,
>       posters,t,
>     } = item;

Reviewer 2: This has some errors in it and as a result is breaking the layout here

<img width="615" alt="Screenshot 2023-11-21 at 4 34 42 PM" src="https://github.com/freeCodeCamp/freeCodeCamp/assets/67210629/b4c6e1ce-d08d-4871-ae2f-d9075a9c1a76">

```suggestion
    const {
      id,
      title,
      views,
      posts_count,
      slug,
      posters,
      category_id,
      bumped_at,
    } = item;
```
